test/e2e/connectivity_test.go

```diff
@@ -121,6 +125,114 @@ func (data *TestData) runPingMesh(t *testing.T, podInfos []PodInfo, ctrname stri
 	}
 }
 
+// runPodsOnWindows checks the Pods' connectivity after antrea-agent is restarted.
+// This case only run on Windows. The Pods in the test includes both host-network Pod and none host-network Pod,
+// because CNI on Windows is also responsible for the host-network Pod's networking as long as it is not using
+// host-process containers.
+func testWindowsPodConnectivityAfterAntreaRestart(t *testing.T, data *TestData) {
+	podInfos, cleanupFn := data.runPodsOnWindows(t)
+	defer cleanupFn()
+
+	data.runPingMesh(t, podInfos, toolboxContainerName, true)
+	nodeFlows := data.countOVSFlowsOnWindows(t, podInfos)
+	err := data.RestartAntreaAgentPods(defaultTimeout)
+	assert.NoError(t, err)
+
+	err = wait.PollUntilContextTimeout(context.Background(), 5*time.Second, 1*time.Minute, false, func(ctx context.Context) (done bool, err error) {
+		newNodeFlows := data.countOVSFlowsOnWindows(t, podInfos)
+		for node, flows := range nodeFlows {
+			newFlows, found := newNodeFlows[node]
+			if !found {
+				return false, nil
+			}
+			if len(newFlows) != len(flows) {
+				return false, nil
+			}
+		}
+		return true, nil
+	})
+	assert.NoErrorf(t, err, "The Openflow entries should be consistent after Antrea agent restarts")
+
+	data.runPingMesh(t, podInfos, toolboxContainerName, true)
+	time.Sleep(10 * time.Minute)
+}
+
+func (data *TestData) dumpOVSFlows(t *testing.T, workerNode string) []string {
+	ovsOfctlCmd := "ovs-ofctl"
+	if clusterInfo.nodesOS[workerNode] == "windows" {
+		ovsOfctlCmd = `c:/openvswitch/usr/bin/ovs-ofctl.exe`
+	}
+	cmd := []string{ovsOfctlCmd, "dump-flows", defaultBridgeName, "--names"}
+	antreaPodName, err := data.getAntreaPodOnNode(workerNode)
+	if err != nil {
+		t.Fatalf("Error when retrieving the name of the Antrea Pod running on Node '%s': %v", workerNode, err)
+	}
+	stdout, stderr, err := data.RunCommandFromPod(antreaNamespace, antreaPodName, ovsContainerName, cmd)
+	if err != nil {
+		t.Fatalf("error when dumping flows: <%v>, err: <%v>", stderr, err)
+	}
+	flows := make([]string, 0)
+	for _, flow := range strings.Split(stdout, "\n") {
+		flow = strings.TrimSpace(flow)
+		if flow == "" {
+			continue
+		}
+		flows = append(flows, flow)
+	}
+	t.Logf("Counted %d flow in OVS bridge '%s' for Node '%s'", len(flows), defaultBridgeName, workerNode)
+	return flows
+}
+
+func (data *TestData) countOVSFlowsOnWindows(t *testing.T, podInfos []PodInfo) map[string][]string {
+	nodeFlows := make(map[string][]string)
+	for idx := range podInfos {
+		workerNode := podInfos[idx].NodeName
+		if _, found := nodeFlows[workerNode]; found {
+			continue
+		}
+		flows := data.dumpOVSFlows(t, workerNode)
+		nodeFlows[workerNode] = flows
+	}
+	return nodeFlows
+}
+
+func (data *TestData) runPodsOnWindows(t *testing.T) ([]PodInfo, func()) {
```

Comment:
as it is currently written, this should not be a separate function IMO. The function has a generic name, but is provisioning Pod in a very specific way, specifically for the needs of the `testWindowsPodConnectivityAfterAntreaRestart` test.

Reply:
updated.